examples/experimental/i3s-17-and-debug/bounding-volume-layer.js

```diff
@@ -78,49 +85,37 @@ export default class BoundingVolumeLayer extends CompositeLayer {
       ...attributes.POSITION,
       value: new Float32Array(attributes.POSITION.value)
     };
+
     for (let i = 0; i < POSITION.value.length; i += 3) {
       const vec = new Vector3(POSITION.value.subarray(i, i + 3));
       vec.add(center);
       POSITION.value.set(vec, i);
     }
-    geometry.attributes = {
-      ...attributes,
-      POSITION
-    };
+    geometry.attributes = {POSITION};
     return geometry;
   }
 
-  _generateMesh(tile, boundingVolumeType) {
-    switch (boundingVolumeType) {
-      case ORIENTED_BOUNDING_BOX:
-        return tile.header.obb ? this._generateCubeMesh(tile) : null;
-      case BOUNDING_SPHERE:
-        return tile.header.mbs ? this._generateSphereMesh(tile) : null;
-      default:
-        return null;
-    }
-  }
-
-  _getBoundingVolumeGeometry(tile, oldLayer, boundingVolumeType) {
-    const mesh = oldLayer && oldLayer.props.mesh;
+  _generateMesh(tile, meshType) {
+    const cubeMesh = tile.header.obb ? this._generateCubeMesh(tile) : null;
+    const sphereMesh = tile.header.mbs ? this._generateSphereMesh(tile) : null;
 
-    if (!mesh || mesh.constructor.name !== boundingVolumeType) {
-      return this._generateMesh(tile, boundingVolumeType);
-    }
+    tile.userData.cubeMesh = cubeMesh;
+    tile.userData.sphereMesh = sphereMesh;
 
-    return mesh;
+    return tile.userData[meshType];
```

Comment:
Maybe put the meshes in a sub-object so that you can type it properly when we introduce typescript.

Reply:
Do you mean mesh specific object inside` tile.userData`?

Reply:
Yeah something like `tile.userData.boundingMeshes = {}`

Reply:
Done